src/Symfony/Component/AutoMapper/AutoMapperInterface.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Symfony\Component\AutoMapper;
+
+/**
+ * Object to object mapper.
+ *
+ * @experimental
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+interface AutoMapperInterface {
+    /**
+     * @param object $object The object to map from
+     * @param null|class-string|object $to The object or class to map to
+     */
+    public function map(object $object, object|string|null $to = null): mixed;
```

Comment:
If `$to` is null, it would guess by looking at the one attribute on `$object` to know what to go to? I'd simplify and require `object|string` - it'll make the code easier to read anyway.

Reply:
And this will allow mapping a class to several others (using the `group` attribute I mentioned in another comment).

Reply:
`$to` is useful only if you already have an `instance` ready to be mapped. A string is handful and yeah it can be optional if you have the `MapTo` attribute. Indeed `object|string $to = null` is enough and it's working like you said?

Reply:
If this allow mapping a class to several others, other attributes may needs more parameter, so we can bind a transform / name mapping to a specific target